website/blog/2025-02-19-react-native-0.78.md

```diff
@@ -0,0 +1,218 @@
+- They have no dimensions. By default, they will display with a 0x0 size and you need to provide a width and height for them to show up.
+- They are not fully customizable at runtime; you can only control dimensions or the overall tint color but you can’t customize individual element attributes _inside_ the resource like stroke widths, border radius, or colors. These types of customizations require different variants of your XML resource.
+
+:::info
+Android’s vector drawables are not a 1:1 replacement for libraries like `react-native-svg`. They are designed specifically for Android and do not work for iOS. If you want to have the same SVGs across all platforms, you'll have to continue using `react-native-svg`. Vector drawables merely offer performance benefits at the expense of customization.
+:::
+
+### ReactNativeFactory on iOS
+
+With React Native 0.78 we improved the integration of React Native on iOS.
+
+This version introduces a new class called `RCTReactNativeFactory` that allows you to create instances of ReactNative without the need of an AppDelegate. This should allow you to create a new version of ReactNative in a ViewController, for example. This simplifies dramatically the integration with Brownfield apps.
+
+Imagine that you want to show a React Native view in a View Controller of your app. Starting from React Native 0.78, what you need to do, after installing all the dependencies as shown in [this guide](https://reactnative.dev/docs/next/integration-with-existing-apps?language=apple#1-set-up-directory-structure), is to add this code:
+
+```diff
+
++import React
++import import React_RCTAppDelegate
+
+public class ViewController {
+
++  var reactNativeFactory: RCTReactNativeFactory?
++  var reactNativeDelegate: ReactNativeDelegate?
+
+  public func viewdidLoad() {
+    super.viewDidLoad()
+    // …
++ reactNativeDelegate = ReactNativeDelegate()
++ reactNativeFactory = RCTReactNativeFactory(delegate: reactNativeDelegate!)
++ view = reactNativeFactory.rootViewFactory.view(withModuleName: "<your module name>")
+  }
+
+}
+
++class ReactNativeDelegate: RCTDefaultReactNativeFactoryDelegate {
+
++  override func sourceURL(for bridge: RCTBridge) -> URL? {
++    self.bundleURL()
++  }
++
++  override func bundleURL() -> URL? {
++    #if DEBUG
++    RCTBundleURLProvider.sharedSettings().jsBundleURL(forBundleRoot: "index")
++    #else
++    Bundle.main.url(forResource: "main", withExtension: "jsbundle")
++    #endif
++  }
++}
+
+```
+
+React Native will be loaded in the View Controller as soon as you navigate to it.
+
+This code creates a `RCTReactNativeFactory`, it assigns a delegate to it and asks it to create a `rootView` for a React Native’s view.
+
+The delegate is defined below and it overrides the `sourceURL` and the `bundleURL` properties to tell React Native where it can find the JS bundle to load in the view.
+
```

Comment:
```suggestion
The delegate is defined below, and it overrides the `sourceURL` and the `bundleURL` properties to tell React Native where it can find the JS bundle to load in the view.

```

Nit.